src/vsr/replica.zig

```diff
@@ -255,7 +260,10 @@ pub fn ReplicaType(
         };
 
         /// Initializes and opens the provided replica using the options.
-        pub fn open(self: *Self, allocator: std.mem.Allocator, options: OpenOptions) !void {
+        pub fn open(self: *Self, parent_allocator: std.mem.Allocator, options: OpenOptions) !void {
+            self.revocable_allocator = RevocableAllocator.init(parent_allocator);
```

Comment:
The Replica should simply accept `allocator` and not try to enforce anything internally, to avoid violating encapsulation. The characteristics of the allocator policy should be decided at a higher layer, by the user.

Reply:
This is the Replica asserting that it doesn't allocate after it finishes init. It seems strange to want assertions about replica behavior to not live in the replica. Are you concerned about a case where someone is using this as a library and wants to allocate using the allocator passed down from replica?

Reply:
> Are you concerned about a case where someone is using this as a library and wants to allocate using the allocator passed down from replica?

Yes, we want to not make the policy decision around type of allocator within Replica itself. That should be the user's prerogative.

Reply:
Your call on whether we can make this work or tradeoffs to balance.

Reply:
The user can still pick the type of allocator. `parent_allocator` can be anything, we just wrap it with an enum that says whether we're allowed to alloc/free.